integration-tests/kubernetes/src/test/java/org/apache/camel/quarkus/component/kubernetes/it/KubernetesDeploymentTest.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.kubernetes.it;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
+import io.fabric8.kubernetes.api.model.apps.DeploymentListBuilder;
+import io.fabric8.kubernetes.client.server.mock.KubernetesServer;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.kubernetes.client.KubernetesTestServer;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.Matchers.is;
+
+@QuarkusTest
+@QuarkusTestResource(CamelQuarkusKubernetesServerTestResource.class)
+class KubernetesDeploymentTest {
```

Comment:
Can any of the test assertions run? Or do we need https://github.com/apache/camel/pull/17140/commits/23b7f03878faf906e81932e2c92fd3dceef666a6 for all of them? If so, we'll just have to disable the test and have an issue to remind us to enable it later.

Reply:
only create and delete work for now, others are affected. Should I `@Disable` the whole class or just temporary wrap the affected methods in some `if`?

Reply:
Wrap in an `if` is fine. You can add a `TODO` comment referencing https://github.com/apache/camel-quarkus/issues/7011 so that we don't forget to update it later.